src/Controls/samples/Controls.Sample/Startup.cs

```diff
@@ -59,23 +64,36 @@ public void Configure(IAppHostBuilder appBuilder)
 						{"Position:Name", "Dictionary_Name" },
 						{"Logging:LogLevel:Default", "Warning"}
 					});
-				})
-				.UseMauiServiceProviderFactory(true)
-				//.UseServiceProviderFactory(new DIExtensionsServiceProviderFactory())
+				});
+
+			if (_pageType == PageType.Blazor)
+            {
+				appBuilder.UseServiceProviderFactory(new DIExtensionsServiceProviderFactory()); // Blazor requires service scopes, which are supported only with Microsoft.Extensions.DependencyInjection
+			}
+			else
+            {
+				appBuilder.UseMauiServiceProviderFactory(constructorInjection: true);
+			}
+
+			appBuilder
 				.ConfigureServices(services =>
 				{
 					services.AddSingleton<ITextService, TextService>();
 					services.AddTransient<MainPageViewModel>();
 
+#if BLAZOR_ENABLED
+					services.AddBlazorWebView();
+#endif
+
 					services.AddTransient(
-						serviceType: typeof(IPage),
+						serviceType: _pageType == PageType.Blazor ? typeof(Page) : typeof(IPage),
 						implementationType: _pageType switch
 						{
 							PageType.NavigationPage => typeof(NavPage),
 							PageType.Xaml => typeof(XamlPage),
 							PageType.Semantics => typeof(SemanticsPage),
 							PageType.Blazor =>
-#if NET6_0_OR_GREATER
+#if BLAZOR_ENABLED
 								typeof(BlazorPage),
 #else
 								throw new NotSupportedException("Blazor requires .NET 6 or higher."),
```

Comment:
It will be interested to see the project template. I deeply hope it doesn't look anything like this file, which is mystifying.

Reply:
The project templates will be .NET 6 only so all the `#if`s go away.